pingcap/tidb-operator
# Evict TiKV leader before resize PVCs

### What problem does this PR solve?


### What is changed and how does it work?


### Code changes


### Tests



1. Deploy cluster with three TiKV.
2. Scale up the storage of TiKV spec.
3. Check the `ComponentVolumeResizing` condition existing in TiKV status.
4. Update the image of TiKV spec, check upgrade is skipped.
5. For three PVCs and Pods, check:
    1. The `tidb.pingcap.com/evict-leader-for-resize: none` annotation in resized Pods is removed 
    1. The `tidb.pingcap.com/evict-leader-for-resize: none` annotation is added to the current Pod.
    2. The storage request is updated of the PVC after the leader count in `status.stores` become 0.
    2. Update the status of the PVC to mock PVC is resized.
    3. Check the `status.volumes` updated in TiKV status and  the `BeginResize` condition existing.
6. After all volume are resized, check all Pods don't have the `tidb.pingcap.com/evict-leader-for-resize: none` annotation.
7. Check the `ComponentVolumeResizing` condition removed in TiKV status.


### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Evict TiKV leader before resize PVCs
```
